Add py314 and drop py39 support

randomir opened the pull request.

randomir committed:
Drop support for Python 3.9

randomir committed:
Add support for Python 3.14

thisac commented on setup.py:
```diff
@@ -66,6 +66,6 @@ def build_extensions(self):
         ],
     install_requires=[
         'numpy>=1.17.3,<3.0.0',  # this is the oldest supported NumPy on Python 3.8
```
Could update this.
```suggestion
        'numpy>=1.21.6,<3.0.0',  # this is the oldest supported NumPy on Python 3.10
```

thisac reviewed (approved):
LGTM

randomir committed:
Update numpy lower bound for py310

Co-authored-by: Theodor Isacsson <theodor@isacsson.ca>

randomir committed:
Update CircleCI macos resource class

`macos.m1.medium.gen1` are being deprecated Feb16 2026, migrating to
`m2pro.medium`.
